Sync/SyncStateMachine.swift

```diff
@@ -104,9 +104,9 @@ open class SyncStateMachine {
     public static let OptimisticStates = Set(SyncStateLabel.optimisticValues)
 
     /// The default set of states that the state machine is allowed to use.
```

Comment:
Oh, interesting approach.  Works in practice?

Reply:
The machinery to restrict the state machine pre-dates this patch, and that's well tested. 

However, does it fix the problem you're concerned about? I haven't been able to verify. @rfk 's script will come in useful here.